src/ComputationContainer/Test/IntegrationTest/ShareTest.hpp

```diff
@@ -1247,4 +1247,106 @@ TEST(ShareTest, FPMulExtraLarge)
     auto rec = recons(a);
 
     EXPECT_EQ(rec[0], (n_parties * 1) * (n_parties * 1));
+}
+
+TEST(ShareTest, unitvPreptest)
+{
+    auto [r, v] = qmpc::Share::unitvPrep<32>();
+    open(v);
+    open(r);
+    auto rec_r = recons(r);
+    auto rec_v = recons(v);
+    rec_r = (rec_r % 32 + 32) % 32;
+    std::cout << "r is " << rec_r << std::endl;
+    EXPECT_EQ(rec_v[rec_r], 1);
+}
+TEST(ShareTest, unitvTest)
+{
+    qmpc::Share::Share<int> n(5);
+    auto w = qmpc::Share::unitv(n);
+    open(w);
+    open(n);
+    auto rec_w = recons(w);
+    auto rec_n = recons(n);
+    std::cout << "n is " << rec_n << std::endl;
+    EXPECT_EQ(rec_w[rec_n % 32], 1);
+}
+
+TEST(ShareTest, expandTest)
+{
+    int x = 5;  // 00000011
+    auto d = qmpc::Share::expand(x);
+    for (auto a : d)
+    {
+        std::cout << a << std::endl;
+    }
```

Comment:
I would like to see it replaced with a comparative test using EXPECT_EQ.

Reply:
ah sure!.
i got it.